libyul/backends/wasm/EVMToEwasmTranslator.cpp

```diff
@@ -524,107 +525,108 @@ function eq(x1, x2, x3, x4, y1, y2, y3, y4) -> r1, r2, r3, r4 {
 }
 
 // returns 0 if a == b, -1 if a < b and 1 if a > b
-function cmp(a, b) -> r {
+function cmp(a, b) -> r:i32 {
 	switch i64.lt_u(a, b)
-	case 1 { r := 0xffffffffffffffff }
+	case 1:i32 { r := 0xffffffff:i32 }
 	default {
 		r := i64.ne(a, b)
 	}
 }
-function lt_320x320_64(x1, x2, x3, x4, x5, y1, y2, y3, y4, y5) -> z {
+function lt_320x320_64(x1, x2, x3, x4, x5, y1, y2, y3, y4, y5) -> z:i32 {
 	switch cmp(x1, y1)
-	case 0 {
+	case 0:i32 {
 		switch cmp(x2, y2)
-		case 0 {
+		case 0:i32 {
 			switch cmp(x3, y3)
-			case 0 {
+			case 0:i32 {
 				switch cmp(x4, y4)
-				case 0 {
+				case 0:i32 {
 					z := i64.lt_u(x5, y5)
 				}
-				case 1 { z := 0 }
-				default { z := 1 }
+				case 1:i32 { z := 0:i32 }
+				default { z := 1:i32 }
 			}
-			case 1 { z := 0 }
-			default { z := 1 }
+			case 1:i32 { z := 0:i32 }
+			default { z := 1:i32 }
 		}
-		case 1 { z := 0 }
-		default { z := 1 }
+		case 1:i32 { z := 0:i32 }
+		default { z := 1:i32 }
 	}
-	case 1 { z := 0 }
-	default { z := 1 }
+	case 1:i32 { z := 0:i32 }
+	default { z := 1:i32 }
 }
-function lt_512x512_64(x1, x2, x3, x4, x5, x6, x7, x8, y1, y2, y3, y4, y5, y6, y7, y8) -> z {
+function lt_512x512_64(x1, x2, x3, x4, x5, x6, x7, x8, y1, y2, y3, y4, y5, y6, y7, y8) -> z:i32 {
 	switch cmp(x1, y1)
-	case 0 {
+	case 0:i32 {
 		switch cmp(x2, y2)
-		case 0 {
+		case 0:i32 {
 			switch cmp(x3, y3)
-			case 0 {
+			case 0:i32 {
 				switch cmp(x4, y4)
-				case 0 {
+				case 0:i32 {
 					switch cmp(x5, y5)
-					case 0 {
+					case 0:i32 {
 						switch cmp(x6, y6)
-						case 0 {
+						case 0:i32 {
 							switch cmp(x7, y7)
-							case 0 {
+							case 0:i32 {
 								z := i64.lt_u(x8, y8)
 							}
-							case 1 { z := 0 }
-							default { z := 1 }
+							case 1:i32 { z := 0:i32 }
+							default { z := 1:i32 }
 						}
-						case 1 { z := 0 }
-						default { z := 1 }
+						case 1:i32 { z := 0:i32 }
+						default { z := 1:i32 }
 					}
-					case 1 { z := 0 }
-					default { z := 1 }
+					case 1:i32 { z := 0:i32 }
+					default { z := 1:i32 }
 				}
-				case 1 { z := 0 }
-				default { z := 1 }
+				case 1:i32 { z := 0:i32 }
+				default { z := 1:i32 }
 			}
-			case 1 { z := 0 }
-			default { z := 1 }
+			case 1:i32 { z := 0:i32 }
+			default { z := 1:i32 }
 		}
-		case 1 { z := 0 }
-		default { z := 1 }
+		case 1:i32 { z := 0:i32 }
+		default { z := 1:i32 }
 	}
-	case 1 { z := 0 }
-	default { z := 1 }
-}
+	case 1:i32 { z := 0:i32 }
+	default { z := 1:i32 }
+}/*
```

Comment:
Haha, this contraption looks quite weird now.

Reply:
Nice, isn't it?